src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -775,8 +787,10 @@ static void create_rr_locked(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
 /* glb_policy->rr_policy may be NULL (initial handover) */
 static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
-  GPR_ASSERT(glb_policy->serverlist != NULL &&
-             glb_policy->serverlist->num_servers > 0);
+  GPR_ASSERT((glb_policy->fallback && glb_policy->fallback_addresses != NULL &&
```

Comment:
Sorry, I wasn't suggesting having both one after the other: I'd move the assert relative to the fallback to the part of the code that's act upon `glb_policy->fallback_addresses`.

Basically, you need to branch depending on whether we are using the serverlist or the fallback addresses. Each branch would have its corresponding version of the assert.